CRFModuleValidation/Schedule.swift

```diff
@@ -0,0 +1,516 @@
+        }).sorted { $0.finishedOn < $1.finishedOn }
+        let json: [String : Any]? = filtered.last?.clientData as? [String : Any]
+        
+        self.hasData = (filtered.count > 0)
+        
+        self.timeOfDay = {
+            guard let tod = json?[Schedule.timeOfDayKey] as? String else { return nil }
+            let components = tod.components(separatedBy: ":")
+            guard components.count == 2, let hour = Int(components[0]), let minute = Int(components[1])
+            else {
+                return nil
+            }
+            
+            var dateComponents = DateComponents()
+            dateComponents.hour = hour
+            dateComponents.minute = minute
+            dateComponents.timeZone = NSTimeZone.default
+            dateComponents.calendar = Calendar.current
+            return dateComponents
+        }()
+        
+        self.daysOfWeek = Schedule.parseDaysOfWeek(scheduleId: scheduleId, daysOfWeek: json?[Schedule.daysOfWeekKey] as? [Int], dayOne: dayOne)
+    }
+    
+    init?(scheduledActivity: SBBScheduledActivity, taskResult:ORKTaskResult, dayOne:Date) {
+        guard let scheduleId = scheduledActivity.taskId,
+            let taskGroup = ScheduleSection.scheduleGroups.first(where: {$0.scheduleTaskIdentifier == scheduleId })
+        else {
+            return nil
+        }
+        
+        self.taskGroup = taskGroup
+        
+        let daysOfWeekResult = taskResult.stepResult(forStepIdentifier: Schedule.daysOfWeekKey)?.results?.first as? ORKChoiceQuestionResult
+        self.daysOfWeek = Schedule.parseDaysOfWeek(scheduleId: scheduleId,
+                                                   daysOfWeek: daysOfWeekResult?.choiceAnswers as? [Int],
+                                                   dayOne: dayOne)
+        
+        let timeOfDayResult = taskResult.stepResult(forStepIdentifier: Schedule.timeOfDayKey)?.results?.first as? ORKTimeOfDayQuestionResult
+        self.timeOfDay = timeOfDayResult?.dateComponentsAnswer
+        self.hasData = true
+    }
+    
+    static func parseDaysOfWeek(scheduleId: TaskIdentifier, daysOfWeek: [Int]?, dayOne:Date) -> Set<Int> {
+        guard let dow = daysOfWeek else {
+//            if scheduleId == .scheduleCheckIn {
+                return Set(1...7)
+//            }
+//            else {
+//                let firstDate = /*(scheduleId == .scheduleWeeklyChallenge) ? dayOne.addingNumberOfDays(1) :*/ dayOne
+//                let dayOfWeek = Calendar.gregorian.component(.weekday, from: firstDate)
+//                return Set([dayOfWeek])
+//            }
+        }
+        return Set(dow)
+    }
+    
+    func clientData() -> SBBJSONValue? {
+        guard let timeDate = self.timeDate else { return nil }
+        let clientData: NSDictionary = [ Schedule.timeOfDayKey : Schedule.timeOfDayFormatter.string(from: timeDate),
+                                         Schedule.daysOfWeekKey : self.daysOfWeek.map({ NSNumber(value: $0) })]
```

Comment:
so... is the client data being set? I am not currently implementing any client data support in ResearchSuite tasks. (when they are finished)